StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,94 @@ export type StoreMessageContentActivityResult = t.TypeOf<
   typeof StoreMessageContentActivityResult
 >;
 
+export const ServicePreferenceError = t.interface({
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference => {
+  const blockedInboxOrChannels = [];
+  if (!servicePreference.isInboxEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.INBOX);
+  }
+  if (!servicePreference.isEmailEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.EMAIL);
+  }
+  if (!servicePreference.isWebhookEnabled) {
+    // eslint-disable-next-line functional/immutable-data
+    blockedInboxOrChannels.push(BlockedInboxOrChannelEnum.WEBHOOK);
+  }
+  return blockedInboxOrChannels;
+};
+
+const getServicePreferenceValueOrError = (
+  servicePreferencesModel: ServicesPreferencesModel
+): ServicePreferenceValueOrError => ({
+  fiscalCode,
+  serviceId,
+  userServicePreferencesMode,
+  userServicePreferencesVersion
+}): TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+> => {
+  if (userServicePreferencesMode === ServicesPreferencesModeEnum.LEGACY) {
+    return fromLeft({
+      kind: "LEGACY",
+      message: "User service preferences mode is LEGACY"
+    });
+  }
+
+  const documentId = makeServicesPreferencesDocumentId(
+    fiscalCode,
+    serviceId,
+    userServicePreferencesVersion as NonNegativeInteger
+  );
+
+  return servicePreferencesModel
+    .find([documentId, fiscalCode])
+    .mapLeft<ServicePreferenceError>(failure => ({
+      kind: "ERROR",
+      message: `COSMOSDB|ERROR=${failure.kind}`
+    }))
+    .map(maybeServicePref =>
+      maybeServicePref.foldL<ReadonlyArray<BlockedInboxOrChannelEnum>>(
+        () =>
+          // if we do not have a preference we return an empty array only
+          // if we have preference mode AUTO, else we must return an array
+          // with BlockedInboxOrChannelEnum.INBOX
+          userServicePreferencesMode === ServicesPreferencesModeEnum.AUTO
+            ? []
+            : [BlockedInboxOrChannelEnum.INBOX],
+        servicePreferenceToBlockedInboxOrChannels
+      )
+    );
+};
+
```

Comment:
```suggestion
> => 
taskEither
    .of<ServicePreferenceError, ServicesPreferencesMode>(
      userServicePreferencesMode
    )
    .chain(
      fromPredicate(
        _ => _ === ServicesPreferencesModeEnum.LEGACY,
        () => ({
          kind: "LEGACY",
          message: "User service preferences mode is LEGACY"
        })
      )
    )
    .map(() =>
      makeServicesPreferencesDocumentId(
        fiscalCode,
        serviceId,
        userServicePreferencesVersion as NonNegativeInteger
      )
    )
    .chain(documentId =>
      servicePreferencesModel
        .find([documentId, fiscalCode])
        .mapLeft<ServicePreferenceError>(failure => ({
          kind: "ERROR",
          message: `COSMOSDB|ERROR=${failure.kind}`
        }))
    )
    .map(maybeServicePref =>
      maybeServicePref.fold<boolean>(
        // if we do not have a preference we return true only
        // if we have preference mode AUTO
        userServicePreferencesMode === ServicesPreferencesModeEnum.AUTO,
        // if we have a preference we return its "isInboxEnabled"
        pref => pref.isInboxEnabled
      )
    );

```

Reply:
Changed and applied in #562062b. This change request had old code that was not applicable directly on github, and an error here:
```
fromPredicate(
        _ => __ === ServicesPreferencesModeEnum.LEGACY_, => **_ !== ServicesPreferencesModeEnum.LEGACY**
        () => ({
          kind: "LEGACY",
          message: "User service preferences mode is LEGACY"
        })
      )
```